In [ ]:
# DATA ANALYSIS

In [30]:
import pandas as pd

In [2]:
data = pd.read_csv('WorldHappiness.csv')
print(data)
print()
print(f"Data shape: {data.shape}")
print()
print(f"Data shape: {data.dtypes}")

     Country name  year  Life Ladder  Log GDP per capita  Social support  \
0     Afghanistan  2008        3.724               7.350           0.451   
1     Afghanistan  2009        4.402               7.509           0.552   
2     Afghanistan  2010        4.758               7.614           0.539   
3     Afghanistan  2011        3.832               7.581           0.521   
4     Afghanistan  2012        3.783               7.661           0.521   
...           ...   ...          ...                 ...             ...   
2358     Zimbabwe  2019        2.694               7.698           0.759   
2359     Zimbabwe  2020        3.160               7.596           0.717   
2360     Zimbabwe  2021        3.155               7.657           0.685   
2361     Zimbabwe  2022        3.296               7.670           0.666   
2362     Zimbabwe  2023        3.572               7.679           0.694   

      Healthy life expectancy at birth  Freedom to make life choices  \
0              

Which countries or regions are the happiest in a particular factor ?

- Economic factors:
    Log GDP per capita
- Social factors:
    Social support
    Generosity
    Perceptions of corruption
- Health Factors:
    Healthy life expectancy at birth
- Psychological and Emotional Factors:
    Life Ladder
    Positive affect
    Negative affect
- Autonomy and Freedom:
    Freedom to make life choices

In [26]:
for column in data.columns[1:]:
    column_mean = data.groupby("Country name")[column].mean()

    country_with_highest_column_mean = column_mean.idxmax()
    highest_column_mean = column_mean.max()

    print(
        f"The country with the highest mean of {column} is {country_with_highest_column_mean} with a mean of {highest_column_mean:.3f}.")

The country with the highest mean of year is Gambia with a mean of 2019.800.
The country with the highest mean of Life Ladder is Denmark with a mean of 7.664.
The country with the highest mean of Log GDP per capita is Luxembourg with a mean of 11.643.
The country with the highest mean of Social support is Iceland with a mean of 0.979.
The country with the highest mean of Healthy life expectancy at birth is Japan with a mean of 73.543.
The country with the highest mean of Freedom to make life choices is Norway with a mean of 0.951.
The country with the highest mean of Generosity is Myanmar with a mean of 0.603.
The country with the highest mean of Perceptions of corruption is Romania with a mean of 0.946.
The country with the highest mean of Positive affect is Panama with a mean of 0.822.
The country with the highest mean of Negative affect is Iraq with a mean of 0.507.


What is the happiest country in each group of factors ?

In [25]:
# ECONOMIC FACTORS

country_mean = data.groupby("Country name")[
    "Log GDP per capita"].mean()

happiest_country = country_mean.idxmax()
happiest_country_total = country_mean.max()

print(
    f"The happiest country based on economic factors is {happiest_country} with an average score of {happiest_country_total:.3f}.")

The happiest country based on economic factors is Luxembourg with an average score of 11.643.


In [24]:
# SOCIAL FACTORS

social_factors = ["Social support", "Generosity", "Perceptions of corruption"]

country_mean = data.groupby("Country name")[social_factors].mean()

country_mean["Total"] = country_mean.mean(axis=1)

happiest_country = country_mean["Total"].idxmax()
happiest_country_total = country_mean["Total"].max()

print(
    f"The happiest country based on social factors is {happiest_country} with an average of {happiest_country_total:.3f}.")

The happiest country based on social factors is Cuba with an average of 0.970.


In [ ]:
# Optionally, print the detailed sums for each country
print("\nDetailed average for each country:")
print(country_mean)

In [23]:
# HEALTH FACTORS

country_mean = data.groupby("Country name")[
    "Healthy life expectancy at birth"].mean()

happiest_country = country_mean.idxmax()
happiest_country_total = country_mean.max()

print(
    f"The happiest country based on health factors is {happiest_country} with an average of {happiest_country_total:.3f}.")

The happiest country based on health factors is Japan with an average of 73.543.


In [22]:
# Psychological and Emotional Factors

social_factors = ["Life Ladder", "Positive affect", "Negative affect"]

country_mean = data.groupby("Country name")[social_factors].mean()

country_mean["Total"] = country_mean.mean(axis=1)

happiest_country = country_mean["Total"].idxmax()
happiest_country_total = country_mean["Total"].max()

print(
    f"The happiest country based on psychological and emotional factors is {happiest_country} with an average of {happiest_country_total:.3f}.")

The happiest country based on psychological and emotional factors is Maldives with an average of 5.198.


In [ ]:
# Optionally, print the detailed sums for each country
print("\nDetailed average for each country:")
print(country_mean)

In [21]:
# Autonomy and Freedom

country_sums = data.groupby("Country name")[
    "Freedom to make life choices"].mean()

happiest_country = country_sums.idxmax()
happiest_country_total = country_sums.max()

print(
    f"The happiest country based on autonomy and freedom factors is {happiest_country} with an average score of {happiest_country_total:.3f}.")

The happiest country based on autonomy and freedom factors is Norway with an average score of 0.951.


What is the happiest country ?

In [27]:
country_mean = data.groupby("Country name")[data.columns[2:]].mean()

country_mean['Average'] = country_mean.mean(axis=1)

happiest_country = country_mean['Average'].idxmax()
highest_total_sum = country_mean['Average'].max()

happiest_country_mean = country_mean.loc[happiest_country, data.columns[2:]]

print(
    f"The happiest country is {happiest_country} with an average of {highest_total_sum:.3f}.")
print()

for column, value in happiest_country_mean.items():
    print(f"{column}: {value:.3f}")

The happiest country is Maldives with an average of 14.441.

Life Ladder: 5.198
Log GDP per capita: 9.893
Social support: 0.913
Healthy life expectancy at birth: 69.775
Freedom to make life choices: 0.855
Generosity: 0.013
Perceptions of corruption: nan
Positive affect: nan
Negative affect: nan


The research indicates that the Maldives is the happiest country because of its high averages in a number of important happiness metrics, including:

Life Ladder: 
A comparatively high score of 5.198 denotes a high level of overall well-being among the populace.

Log of GDP per person: 
a robust economy with an average score of 9.893, indicating a comparatively good standard of living.

Social Support: 
High social networks and community support, which are essential for happiness, are indicated by an average score of 0.913.

Expectancy of a Healthy Life at Birth: 
a high average age of 69.775 years, which is indicative of decent living and healthcare circumstances.

Freedom to Choose in Life: 
a noteworthy degree of individual autonomy with an average score of 0.855.

The Maldives is the happiest country in this dataset, despite the fact that the data for "Perceptions of corruption," "Positive affect," and "Negative affect" are lacking (NaN). This is due to the country's outstanding performance in the categories that are available. One possible explanation for this high rating is the lack of negative evidence, such as high opinions of corruption.
